rclpy/executors/examples_rclpy_executors/custom_executor.py

```diff
@@ -45,16 +46,14 @@ class PriorityExecutor(Executor):
     def __init__(self):
         super().__init__()
         self.high_priority_nodes = set()
-        self.low_priority_thread = None
+        self.hp_executor = ThreadPoolExecutor(max_workers=multiprocessing.cpu_count())
```

Comment:
`cpu_count()` may raise `NotImplementedError` which you need to catch and fall back to a hard coded value.

Reply:
How about omitting max_workers? [In python 3.5 it will use os.cpu_count() * 5 if it's implemented, or 5 workers if it's not.](https://github.com/python/cpython/blob/0d68d6d9c6386e3f78893f0506f2ce16d88b5db8/Lib/concurrent/futures/thread.py#L91-L94)

Reply:
Sounds like high defaults when not doing IO-intense operations. Since this example doesn't "do" much it probable doesn't matter. Not sure if it sets a good example - maybe with a comment to document what the default is that is ok.

Reply:
cpu_count() or 4 workers in 5faf21687bb49ee6889b8bf48de6d38532456ee5